### PR TITLE
Translations cleanup: plural syntax improvements

### PR DESCRIPTION
**Pull request by vincentbpro**
### What does it do?

- Adds tests in in translation dirs that fails when the .singular/.plural syntax is used in translations
- Turns the .singular/.plural syntax into [formatjs templates](https://formatjs.io/docs/core-concepts/icu-syntax#plural-format)

### Why is it needed?

Helps maintaining translations by enforcing a standard and improving consistency

### CODE REVIEW

**Review by ronronscelestes — APPROVED** [review]
LGTM 🔥

File: packages/core/content-type-builder/admin/src/translations/en.json [hunk 1]
```diff
@@ -133,14 +133,13 @@
...
   "from": "from",
   "listView.headerLayout.description": "Build the data architecture of your content",
-  "menu.section.components.name.plural": "Components",
-  "menu.section.components.name.singular": "Component",
-  "menu.section.models.name.plural": "Collection Types",
-  "menu.section.models.name.singular": "Collection Type",
-  "menu.section.single-types.name.plural": "Single Types",
-  "menu.section.single-types.name.singular": "Single Type",
+  "menu.section.components.name": "Components",
+  "menu.section.models.name": "Collection Types",
```
Comment (soupette) [inline packages/core/content-type-builder/admin/src/translations/en.json:141]: I think we need to delete the corresponding keys in the other language can you run the cleaning script?

Comment (vincentbpro) [reply packages/core/content-type-builder/admin/src/translations/en.json:141]: Sure, doing this rn. Just wanted to do keep the diff readable :)

Comment (vincentbpro) [reply packages/core/content-type-builder/admin/src/translations/en.json:141]: @soupette done!

**Review by soupette — DISMISSED** [review]

**Review by soupette — APPROVED** [review]

**Pull request merged**